### DISCUSSION TITLE
Docs: Simplifying the installation page.

### DISCUSSION THREAD

**Discussion by artofcomputing** [apache/superset#25976] [Ideas]
**A little bit of context**: A while ago on the documentation bash calls, we discussed the idea of changing the current installation method to make easier for users to install and use Superset right away, simplifying the installation method to mirror the [Docker page](https://hub.docker.com/r/apache/superset). 

This has a few benefits, these being:
- Simplifying the installation method.
- Allowing users to get up and running a Superset instance faster.

While allowing us to:
- Split installation and configuration topics.
- Provide a stable playground for users to explore (by making the user to install a stable release of Superset).

So what does everyone think? Would mirroring the Docker page for installing Superset be a viable option? What are your thoughts on the subject? Any feedback or points of interest?

**Comment by sfirke** [comment]
Would this be the structure that Airflow has?  Check out their table of contents on the left side of the docs.

First there's "Quick Start" which is a few steps to get running with a local pip install version.  I think that's fine, though I'm not sure how useful that will be given that (I think) most people want to run Superset remotely.

Next is their Installation of Airflow page which I love and think we should imitate.  It describes the different ways to install: PyPI, Docker, Kubernetes Helm, etc. and lists their pros and cons.

We have the different subpages for Docker, PyPI ("from scratch") but not either of these broader pages.  I would be in favor of adding both of them, with the comparison page ("Installing Superset") being the top priority.

Is that what you're suggesting or am I misunderstanding?

**Reply by artofcomputing** [reply]
Pretty much spot on @sfirke, we could start with the same format Airflow has, since having a meta installation page would make the life of Superset users much easier. 

I think that a Quick Start is extremely valuable to get users up and running a playground instance of Superset (to get accustomed with dash design/queries/so on), but it should be just that: a playground to explore. With that we can split between exploration of Superset, and a full-fledged production deployment of it on specific installation pages. There's also a few other ASF Projects that follow that road:
- https://kafka.apache.org/quickstart
- https://cassandra.apache.org/_/quickstart.html
- https://nightlies.apache.org/flink/flink-docs-stable/docs/try-flink/local_installation/

After the Quick Start we can direct users with a What's Next for them to check out how to create dashes/configure DBs, and the meta installation page.

**Reply by artofcomputing** [reply]
Following the docs hierarchy, we could organize it like this:

- Quickstart
   - What's Next? (points to the Installation meta page/dataviz pages)
- Installation of Superset
   - Install Methods
- Dashboards/Charts
- etc

**Reply by sfirke** [reply]
All of that sounds great.  I get what you're saying and agree, a nice benefit would be if we can make the quickstart approach have none of the security hassles.  Just dreaming out loud here: there could be a variable that indicates it's a playground instance and if true, it (a) disables a lot of security (b) displays a banner message or startup message or similar saying "test instance" or something.

**Reply by artofcomputing** [reply]
That makes sense, a bit off-topic but IMO the SECRET_KEY should really be something automatically generated each time you start a fresh new instance and persists until rotated by a predefined date/time, that way users wouldn't even think about it, and could manually set it based on their needs.

Anyway, I'll start forking and start making some changes. I'll use the Docker Quickstart as a template and we can go from there.

**Reply by artofcomputing** [reply]
Made a simple Quickstart page and organized the docs pages at my branch https://github.com/artofcomputing/superset/tree/feat/quickstart

This is what the page would look like
<img width="1323" alt="Screenshot 2023-11-20 at 21 33 28" src="https://github.com/apache/superset/assets/45772806/9ec95010-f20c-43ce-9460-ac333719a7b7">

I made the SUPERSET_SECRET_KEY use the openssl rand -base64 42 command to automatically generate a secret key, I think it would be good to add a tip after the command pointing to why it's important, like it was done for the Docker Compose install.

It's missing the links at the end for the pages after What's Next, I'll add that later and point to the meta installation page too.

**Reply by artofcomputing** [reply]
Made a few adjustments on the documentation structure, and added tips about the secret key flag. Also I took the instruction numbers down to just 5 instructions, that way is easier for new users to follow the Quickstart.

![image](https://github.com/apache/superset/assets/45772806/85cc0ccb-159f-437a-8092-676dcc407465)
![image](https://github.com/apache/superset/assets/45772806/1ee5c0f7-e0b9-4144-8bdc-0ae0abf3b575)

**Reply by nytai** [reply]
Should we make a note about this using a SQLite db and data will be lost of the container exits? Lots of users have found this out the hard way when following the instructions on the dockerhub page

**Reply by artofcomputing** [reply]
That's a good question @nytai, I'm not sure how to frame that though, could be something like this?

<img width="863" alt="image" src="https://github.com/apache/superset/assets/45772806/a9ce3094-36f1-47a7-a263-0c1055825db3">

**Comment by artofcomputing** [comment]
PR #26129 created with Feedback provided.
